src/Shared/InterningBinaryReader.cs

```diff
@@ -146,15 +150,25 @@ override public String ReadString()
                         charsRead = _decoder.GetChars(rawBuffer, rawPosition, n, charBuffer, 0);
                         return Strings.WeakIntern(charBuffer.AsSpan(0, charsRead));
                     }
-
+#if !NET35
+                    resultBuffer ??= ArrayPool<char>.Shared.Rent(stringLength); // Actual string length in chars may be smaller.
```

Comment:
I'm guessing that if some other process is already renting this, it allocates another one? Maybe another interesting case in which parallelism reduces efficiency.

Reply:
That is correct. However, degree of parallelism in this particular test case is small. Only two buffers has been allocated in devenv and 1 per each msbuild node. Measured by `*System.Buffers.ArrayPoolEventSource` ETW.

Reply:
Note that it's note per _process_; it'd have to be threads. The ArrayPool is process-local. I'm curious what caused the second one to be allocated here since I thought the IPC stuff was almost entirely single-threaded.